### PR TITLE
feat: [CDS-36210]: move Artifactory services and related to lower modules

### PR DESCRIPTION
**Pull request by abhinavmittal-harness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32518)

### CODE REVIEW

File: 960-api-services/src/main/java/software/wings/common/BuildDetailsComparatorAscending.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import io.harness.annotations.dev.TargetModule;
+import io.harness.artifact.ComparatorUtils;
+
+import software.wings.helpers.ext.jenkins.BuildDetails;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+@OwnedBy(CDC)
+@TargetModule(HarnessModule._960_API_SERVICES)
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/software/wings/common/BuildDetailsComparatorAscending.java:23]: Remove target module

File: 957-cg-beans/src/main/java/software/wings/beans/artifact/ArtifactFile.java [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+import io.harness.beans.EmbeddedUser;
+import io.harness.delegate.beans.ChecksumType;
+
+import software.wings.beans.BaseFile;
+
+/**
+ * The Class ArtifactFile.
+ */
+@OwnedBy(CDC)
+@TargetModule(HarnessModule._957_CG_BEANS)
```
Comment (puthrayaharness) [inline 957-cg-beans/src/main/java/software/wings/beans/artifact/ArtifactFile.java:24]: Remove target module

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Review by raghuAtWings — COMMENTED** [review]
> Reviewed 25 of 44 files at r1, 10 of 16 files at r2, 9 of 9 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32518)* status: 1 change requests. 0 of 1 approvals obtained.

**Review by raghuAtWings — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32518)* status: :shipit: complete! ✓

**Review by abhinavmittal-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32518)* status: 12 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/serializer/kryo/DelegateTasksKryoRegistrar.java` line 356 at r3](https://reviewable.io/reviews/harness/harness-core/32518#-N1Jow-T7CWf714wB3TM:-N1Kvi834FmMKbJb90eC:b-omr9u7) ([raw file](https://github.com/harness/harness-core/blob/1c1d2c39be66b477bc73546cbe3cfbacdc32ab49/930-delegate-tasks/src/main/java/io/harness/serializer/kryo/DelegateTasksKryoRegistrar.java#L356)):*
<details><summary><i>Previously, raghuAtWings (Raghvendra Singh) wrote…</i></summary><blockquote>

Where is this added?
</blockquote></details>

957 cg beans
___
*[`957-cg-beans/src/main/java/software/wings/beans/artifact/ArtifactFile.java` line 24 at r2](https://reviewable.io/reviews/harness/harness-core/32518#-N0ZoRfAbsIkS40o9CHI-r2-24:-N1KvkCnELwTyEdmWUaV:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/9c87eae8a343a16ff901a9dc88c9fdf837f090f2/957-cg-beans/src/main/java/software/wings/beans/artifact/ArtifactFile.java#L24)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

Remove target module
</blockquote></details>

Done.
___
*[`960-api-services/src/main/java/software/wings/common/BuildDetailsComparatorAscending.java` line 23 at r2](https://reviewable.io/reviews/harness/harness-core/32518#-N0ZoRfAbsIkS40o9CHJ-r2-23:-N1KvkiDB6UFQ_zXrzBW:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/9c87eae8a343a16ff901a9dc88c9fdf837f090f2/960-api-services/src/main/java/software/wings/common/BuildDetailsComparatorAscending.java#L23)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

Remove target module
</blockquote></details>

Done.

**Review by raghuAtWings — DISMISSED** [review]
> Reviewed 12 of 12 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32518)* status: 1 change requests. 2. unresolved discussions.

**Review by raghuAtWings — APPROVED** [review]
> Reviewed 5 of 5 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32518)* status: :shipit: complete! ✓

**Pull request merged**